src/fmu/dataio/dataio.py

```diff
@@ -134,12 +146,12 @@ def _check_content(proposed: Union[str, dict]) -> Any:
         )
 
     logger.debug("outgoing content is set to %s", usecontent)
-    if useextra:
-        _content_validate(usecontent, useextra)
-        return {usecontent: useextra}
+    if content_specific:
+        _content_validate(usecontent, content_specific)
     else:
         logger.debug("content has no extra information")
-        return usecontent
+
+    return usecontent, content_specific
```

Comment:
This is the key change. Instead of returning only the content (which can be a string or a dict), function now returns 1) the string that shall become `data.content` and the dict that shall become `data.<content>`. If this is not relevant, the latter is `None` but still returned.